scripts/jenkins/code-coverage.groovy

```diff
@@ -17,58 +14,72 @@ pipeline {
+                    hostname
+                    pwd
+                    yum install -y httpd lcov
+                    systemctl start httpd.service
+                    make uninstall || true
+                    make clean || true 
+                    rm -rf cortx-motr
                     OLD_BUILD_NUMBER=$(echo "$BUILD_NUMBER - 10" | bc)
                     echo "Old build number is : $OLD_BUILD_NUMBER"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "rm -rf /var/www/html/$OLD_BUILD_NUMBER || true"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/ ; git clone --recursive -b ${BRANCH} https://github.com/Seagate/cortx-motr"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; sudo ./scripts/install-build-deps"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; ./autogen.sh"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; ./configure --enable-coverage --enable-debug ${OPTIONS}"
-                    sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; make -j6"
+                    rm -rf /var/www/html/$OLD_BUILD_NUMBER || true
+                    git clone --recursive -b ${BRANCH} https://github.com/Seagate/cortx-motr  cortx-motr
+                    cd cortx-motr
+                    sudo ./scripts/install-build-deps
+                    ./autogen.sh
+                    ./configure --enable-coverage --enable-debug ${OPTIONS}
+                    make -j6
 
                     if [ ${UT} == true ]
                     then
-                       sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; sudo ./scripts/m0 run-ut"
+                       ./scripts/m0 run-ut -l > "$TestList"
+                       set +o errexit
+                       while read Test
+                       do
+                               echo "Executing: $Test"
+                               ./scripts/m0 run-ut -t $Test 
+                               echo "Test $Test Executed and exited with return code $?"
+                       done < "$TestList"
                     fi
 
                     if [ ${ST} == true ]
                     then
-                        sshpass -p ${NODE_PASS} ssh -o StrictHostKeyChecking=no ${NODE_USER}@${NODE_HOST} "cd /root/cortx-motr ; sudo ./scripts/m0 run-st"
+                       ./scripts/m0 run-ut -l > "$TestList"
+                       set +o errexit
+                       while read Test
+                       do
+                               echo "Executing: $Test"
+                               ./scripts/m0 run-st -t $Test 
```

Comment:
We need a timeout and cleanup here for the cases where ST might be stuck.

Reply:
I have added the timeout code as per your suggestion. 
As for cleanup we are doing it at the start instead of doing at the end, so that it is easy for us to debug.